doc/logging.md

```diff
@@ -0,0 +1,24 @@
+# How does Happy test log work
+
+## Types of Happy logs
+  * Happy host logs — Happy will run as host, Happy host logs will have logs from Happy tool only.
+
+  * Happy device logs — Happy will host virtual network which has multiple virtual devices, Happy device logs will have logs for software running on the virtual device, openweave is one of the software example, so device logs will have openweave logs.
+
+## Where Happy logs are stored
+
+Both Happy host and device logs location are processed in `happy/Driver.py` as below:
+  * If `HAPPY_LOG_DIR` is found in os environment variable, Happy uses `HAPPY_LOG_DIR` value as log folder location to store Happy test logs. For example, openweave-core tests store Happy test logs along with feature folder under build folder.
+  * If `HAPPY_LOG_DIR` is not found in os environment variable, Happy uses `default_happy_log_dir` from [happy/conf/main_config.json](https://github.com/openweave/happy/blob/master/happy/conf/main_config.json) as the default log folder location to store Happy test logs.
+
+## How to set `HAPPY_LOG_DIR`
+  There is multiple ways to set `HAPPY_LOG_DIR` in os environment variable, see examples below:
+  * In openweave test, we add `HAPPY_LOG_DIR=<dir value>` in command line along with the test script we want to run in [weave-parallel-test-engine](https://github.com/openweave/openweave-core/blob/master/src/test-apps/weave-parallel-test-engine)
+  * We can add `HAPPY_LOG_DIR=<dir value>` in .bashrc, and source it before we run the test script.
+
+## Where is logs location variable being used
+* Happy host logs location variable are used by python logging module to store Happy host logs in [Driver.py](https://github.com/openweave/happy/blob/master/happy/Driver.py#L223)
```

Comment:
Do users of Happy need to know this or do developers and maintainers of Happy need to know this?

The documentation seems to co-mingle information for users and developers. However, the documentation should keep those roles and concerns separate. The user wants to know "How do I just use Happy and get my job done with it?". The developer wants to know, "How do I maintain and extend Happy  to accomplish a new task or implement a new feature?".

Reply:
Users don't need to know this part information, how about I delete this part. We can think about do we need to keep the information somewhere and where to keep.